clippy_lints/src/unnecessary_map_on_constructor.rs

```diff
@@ -9,19 +9,22 @@ use rustc_span::sym;
 
 declare_clippy_lint! {
     /// ### What it does
-    /// Suggest removing the use of a map (or map_err) method when an Option or Result is being constructed.
+    /// Suggests removing the use of a `map()` (or `map_err()`) method when an `Option` or `Result`
+    ///  is being constructed.
     ///
     /// ### Why is this bad?
     /// It introduces unnecessary complexity. In this case the function can be used directly and
-    /// construct the Option or Result from the output.
+    /// construct the `Option` or `Result` from the output.
```

Comment:
I think it'd be good to reword the second sentence a bit here, maybe something like

> In this case the function can be used directly in the `Some`, `Ok` or `Err` constructors.

Reply:
I put in a different new wording:

> It introduces unnecessary complexity. Instead, the function can be called before
> constructing the `Option` or `Result` from its return value.